### PR TITLE
Enable linux perf runs

### PR DESCRIPTION
**Pull request by jorive**
- Update the xUnit Performance package with a bug fixed that was silently making the tests fail
- Removed the `[MeasureInstructionsRetired]` attribute from the benchmarks source code, as it's not use, the reason the tests were failing, and this is now passed on the command line when profiling the tests.
- Updated the `run-xunit-perf.sh` and `perf.groovy` files with the appropriate changes.

### CODE REVIEW

File: tests/scripts/run-xunit-perf.sh [hunk 1]
```diff
@@ -333,69 +283,96 @@ if [ ! -d "$testRootDir" ]; then
...
+        cp "$directory/$filename"*.txt .  || exit 1
+    fi
+
+    # TODO: Do we need this here.
+    chmod u+x ./corerun
+
+    echo "----------"
+    echo "  Running $testname"
+    echo "----------"
+    run_command $stabilityPrefix ./corerun PerfHarness.dll $test --perf:runid Perf --perf:collect stopwatch || exit 1
```
Comment (dpodder) [inline tests/scripts/run-xunit-perf.sh]: This hardcodes `--perf:collect` and `stopwatch`, ignoring the input parameters passed to this script. Is that intentional?

**Comment by AndyAyersMS** [comment]
It feels wrong to me to remove the `[MeasureInstructionsRetired]` annotations -- wouldn't it have been better to fix xunit-perf to handle the exception that this causes on Linux and / or fix whatever the underlying issue is that causes the exception?

However, I'm not going to hold up the PR over this, since I'd like to get the Linux measurements unblocked.

Are you able to trigger perf runs pre-merge and verify they produce the expected data, or are we held hostage to lagging visibility of changes to perf.groovy?

**Comment by jorive** [comment]
@AndyAyersMS `[MeasureInstructionsRetired]` is fixed in the xUnit version that was also updated in this PR. In addition, this is still measure in the profile=on runs, and disabled on profile=off, so you still get the `Instructions Retired` results through PR and rolling runs.

**Comment by jorive** [comment]
@AndyAyersMS I was able to run the benchmarks offline, and I am in the process to add a new machine to Jenkins (Old machine hardware died), so I can tests the E2E scenario and validate.

**Comment by jorive** [comment]
@dotnet-bot test Ubuntu14.04 perf
@dotnet-bot test Windows_NT x64 perf
@dotnet-bot test Windows_NT x86 perf

**Comment by jorive** [comment]
@dotnet-bot test Ubuntu14.04 perf

**Comment by DrewScoggins** [comment]
test ci please

**Comment by jorive** [comment]
@dotnet-bot test Ubuntu14.04 perf
@dotnet-bot test Windows_NT x64 perf
@dotnet-bot test Windows_NT x86 perf

**Comment by jorive** [comment]
@dotnet-bot test Ubuntu14.04 perf
@dotnet-bot test Windows_NT x64 perf
@dotnet-bot test Windows_NT x86 perf

**Comment by DrewScoggins** [comment]
test ci please

**Comment by DrewScoggins** [comment]
test ci please

**Comment by jorive** [comment]
test ci please

**Comment by jorive** [comment]
test ci please

**Comment by jorive** [comment]
@DrewScoggins I just set the maximum number of iterations to match the Windows runs too.

**Pull request merged**